src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -2205,7 +2205,7 @@ PropertyPath Visitor::visit(Parser::PathNegatedPropertySetContext* ctx) {
 PropertyPath Visitor::visit(Parser::PathOneInPropertySetContext* ctx) {
   auto iri = ctx->iri() ? visit(ctx->iri()) : a;
   const std::string& text = ctx->getText();
-  AD_CORRECTNESS_CHECK((iri == a) == (text == "a" || text == "^a"));
+  AD_CORRECTNESS_CHECK(iri == a || !(text == "a" || text == "^a"));
```

Comment:
More in the spirit of the (rather technical) assertion IMHO would be:
```suggestion
  AD_CORRECTNESS_CHECK((static_cast<bool>(ctx->iri()) == !(text == "a" || text == "^a"));
```

(If the `IRI` rule doesn't match, then and only then we have the special `a` or `^a` rules.
As I understand this assertion is because we parse the rule not exactly as written in the grammar, so we want to guard against accidental errors in this rewrite.